http/handler.go

```diff
@@ -115,7 +115,7 @@ func respondStandby(core *vault.Core, w http.ResponseWriter, reqURL *url.URL) {
 	redirectURL := url.URL{
 		Scheme:   advertiseURL.Scheme,
 		Host:     advertiseURL.Host,
-		Path:     reqURL.Path,
+		Path:     advertiseURL.Path + reqURL.Path,
```

Comment:
```
import path
path.Join(advertiseURL.Path, reqURL.Path)
```

There is probably also a function for URIs as well, would have to check.

Just please remember - we need to upstream it at some point. You can also leave it as-it and I will clean it up/upstream it while working on bumping Vault to something newer.